packages/@react-aria/slider/src/useMultiSlider.ts

```diff
@@ -0,0 +1,105 @@
+  trackProps: HTMLAttributes<HTMLElement>;
+
+  /** Label ID for labling slider inputs. Should be passed on in `useSliderThumb`. */
+  labelId: string;
+}
+
+/**
+ * Provides behavior and accessibility for a slider component.
+ * 
+ * @param props Props for the slider.
+ * @param state State for the slider, as returned by `useMultiSliderState`.
+ * @param trackRef Ref for the "track" element.  The width of this element provides the "length"
+ * of the track -- the span of one dimensional space that the slider thumb can be.  It also
+ * accepts click and drag motions, so that the closest thumb will follow clicks and drags on
+ * the track.
+ */
+export function useMultiSlider(
+  props: MultiSliderProps, 
+  state: MultiSliderState, 
+  trackRef: React.RefObject<HTMLElement>
+): MultiSliderAria {
+  const {labelProps, fieldProps} = useLabel(props);
+
+  // When the user clicks or drags the track, we want the motion to set and drag the 
+  // closest thumb.  Hence we also need to install useDrag1D() on the track element.
+  // Here, we keep track of which index is the "closest" to the drag start point.
+  // It is set onMouseDown; see trackProps below.
+  const realTimeTrackDraggingIndex = useRef<number | undefined>(undefined);
+  const draggableProps = useDrag1D({
+    containerRef: trackRef as any,
+    reverse: false,
+    orientation: 'horizontal',
+    onDrag: (dragging) => {
+      if (realTimeTrackDraggingIndex.current !== undefined) {
+        state.setDragging(realTimeTrackDraggingIndex.current, dragging);
+      }
+    },
+    onPositionChange: (position) => {
+      if (realTimeTrackDraggingIndex.current !== undefined) {
+        state.setValue(realTimeTrackDraggingIndex.current, computeOffsetToValue(position, props, trackRef));
+      }
+    }
+  });
+
+  return {
+    labelProps,
+
+    // The root element of the Slider will have role="group" to group together
+    // all the thumb inputs in the Slider.  The label of the Slider will
+    // be used to label the group.
+    containerProps: {
+      role: 'group',
+      ...fieldProps
+    },
+    trackProps: mergeProps({
+      onMouseDown: (e: React.MouseEvent<HTMLElement>) => {
+        // We only trigger track-dragging if the user clicks on the track itself.
+        if (trackRef.current && trackRef.current === e.target) {
+          // Find the closest thumb
+          const trackPosition = trackRef.current.getBoundingClientRect().left;
+          const clickPosition = e.clientX;
+          const offset = clickPosition - trackPosition;
+          const value = computeOffsetToValue(offset, props, trackRef);
+          const minDiff = Math.min(...state.values.map(v => Math.abs(v - value)));
+          const index = state.values.findIndex(v => Math.abs(v - value) === minDiff);
+          if (index >= 0) {
+            // Don't unfocus anything
+            e.preventDefault();
+
+            realTimeTrackDraggingIndex.current = index;
+            state.setFocusedIndex(index);
```

Comment:
Do you think we should also set the value of the slider here as well? Currently, if you mouse down on the track without moving your mouse, the thumb does not move. It only moves after you drag the mouse.